op-service/oppprof/service.go

```diff
@@ -0,0 +1,139 @@
+package oppprof
+
+import (
+	"context"
+	"io"
+	"net"
+	"net/http"
+	httpPprof "net/http/pprof"
+	"os"
+	"path/filepath"
+	"runtime"
+	"runtime/pprof"
+	"strconv"
+
+	"github.com/ethereum-optimism/optimism/op-service/httputil"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type Service struct {
+	listenEnabled bool
+	listenAddr    string
+	listenPort    int
+
+	profileType     string
+	profileDir      string
+	profileFilename string
+
+	cpuFile    io.Closer
+	httpServer *httputil.HTTPServer
+}
+
+func New(listenEnabled bool, listenAddr string, listenPort int, profileType, profileDir, profileFilename string) *Service {
+	return &Service{
+		listenEnabled:   listenEnabled,
+		listenAddr:      listenAddr,
+		listenPort:      listenPort,
+		profileType:     profileType,
+		profileDir:      profileDir,
+		profileFilename: profileFilename,
+	}
+}
+
+func (s *Service) Start() error {
+	switch s.profileType {
+	case "cpu":
+		if err := s.startCPUProfile(); err != nil {
+			return err
+		}
+	case "block":
+		runtime.SetBlockProfileRate(1)
+	case "mutex":
+		runtime.SetMutexProfileFraction(1)
+	}
+	if s.listenEnabled {
+		if err := s.startServer(); err != nil {
+			return err
+		}
+	}
+	if s.profileType != "" {
+		log.Info("start profiling to file", "profile_type", s.profileType, "profile_filepath", s.buildTargetFilePath())
+	}
+	return nil
+}
+
+func (s *Service) Stop(ctx context.Context) error {
+	switch s.profileType {
+	case "cpu":
+		pprof.StopCPUProfile()
+		if s.cpuFile != nil {
+			if err := s.cpuFile.Close(); err != nil {
+				return err
+			}
+		}
+	case "heap":
+		runtime.GC()
+		fallthrough
+	default:
+		profile := pprof.Lookup(s.profileType)
+		if profile == nil {
+			break
+		}
+		filepath := s.buildTargetFilePath()
+		log.Info("saving profile info", "profile_type", s.profileType, "profile_filepath", s.buildTargetFilePath())
+		f, err := os.Create(filepath)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		_ = profile.WriteTo(f, 0)
+	}
+	if s.httpServer != nil {
+		if err := s.httpServer.Stop(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *Service) startServer() error {
+	log.Debug("starting pprof server", "addr", net.JoinHostPort(s.listenAddr, strconv.Itoa(s.listenPort)))
+	mux := http.NewServeMux()
+
+	// have to do below to support multiple servers, since the
+	// pprof import only uses DefaultServeMux
+	mux.Handle("/debug/pprof/", http.HandlerFunc(httpPprof.Index))
+	mux.Handle("/debug/pprof/cmdline", http.HandlerFunc(httpPprof.Cmdline))
```

Comment:
I see this is copied over, but I wonder if we should disable this option. We pass potentially sensitive values via the command line (most pertinently the L1 RPC url which could have an embedded API key, the L2 engine JWT secret is hidden b/c that value is passed in via a file).

Reply:
Point seems valid. But if we do, we break backwards compatibilty. It seems like a small change, but I can imagine someone relying on this. Should we create another small issue and mark that it breaks compatibility?

Reply:
A tough dilemma. IMO it's better to break backwards compatibility vs. allowing the potential "footgun" of accidentally leaking sensitive values.

Reply:
We can try to filter sensitive data via middleware. But it's risky if some new sensitive flag is added and we forgot to filter it out here. Removed this handler completely by now.

Reply:
Can you help, which branch should I select as target now? There are multiple release/ branches and Active Branches table in docs does not contain any of them.

Reply:
Use develop as the target branch